Doc/library/io.rst

```diff
@@ -64,6 +64,13 @@ In-memory text streams are also available as :class:`StringIO` objects::
 
    f = io.StringIO("some initial text data")
 
+.. note::
+   If you are working with a non-blocking stream, be aware that operations on text I/O
+   objects may raise a ``BlockingIOError``. This occurs when the underlying stream is
+   in non-blocking mode and a read operation cannot be completed immediately, potentially
```

Comment:
I think the `, potentially leading to a ``BlockingIOError``.` part can be dropped in this sentence / is redundant with the "may raise" part of the sentence earlier on.

Reply:
Thank you for your feedback and suggestions. I found them very helpful, especially since I don't have daily practice with writing standard library documentation. I've made changes based on your recommendations and addressed all the points you raised. I hope these adjustments align better with the standard library style now.